examples/tls_server_context_wolfssl.cc

```diff
@@ -0,0 +1,302 @@
+              << std::endl;
+    return -1;
+  }
+
+  if (ngtcp2_crypto_wolfssl_configure_server_context(ssl_ctx_) != 0) {
+    std::cerr << "ngtcp2_crypto_openssl_configure_server_context failed"
+              << std::endl;
+    return -1;
+  }
+
+#ifdef WOLFSSL_EARLY_DATA
+  wolfSSL_CTX_set_max_early_data(ssl_ctx_, UINT32_MAX);
+#endif
+
+  constexpr auto ssl_opts = (WOLFSSL_OP_ALL & ~WOLFSSL_OP_DONT_INSERT_EMPTY_FRAGMENTS) |
+                            WOLFSSL_OP_SINGLE_ECDH_USE |
+                            WOLFSSL_OP_CIPHER_SERVER_PREFERENCE;
+
+  wolfSSL_CTX_set_options(ssl_ctx_, ssl_opts);
+
+  if (wolfSSL_CTX_set_cipher_list(ssl_ctx_, config.ciphers) != 1) {
+    std::cerr << "wolfSSL_CTX_set_ciphersuites: "
+              << ERR_error_string(ERR_get_error(), nullptr) << std::endl;
+    return -1;
+  }
+
+  /* Currently not supported.
+  if (wolfSSL_CTX_set1_curves_list(ssl_ctx_, (char*)config.groups) != 1) {
+    std::cerr << "wolfSSL_CTX_set1_curves_list(" << config.groups << ") failed" << std::endl;
+    return -1;
+  }
+  */
+
+  wolfSSL_CTX_set_mode(ssl_ctx_, SSL_MODE_RELEASE_BUFFERS);
+
+  switch (app_proto) {
+  case AppProtocol::H3:
+    wolfSSL_CTX_set_alpn_select_cb(ssl_ctx_, alpn_select_proto_h3_cb, nullptr);
+    break;
+  case AppProtocol::HQ:
+    wolfSSL_CTX_set_alpn_select_cb(ssl_ctx_, alpn_select_proto_hq_cb, nullptr);
+    break;
+  case AppProtocol::Perf:
+    wolfSSL_CTX_set_alpn_select_cb(ssl_ctx_, alpn_select_proto_perf_cb, nullptr);
+    break;
+  }
+
+  wolfSSL_CTX_set_default_verify_paths(ssl_ctx_);
+
+  if (wolfSSL_CTX_use_PrivateKey_file(ssl_ctx_, private_key_file,
+                                     SSL_FILETYPE_PEM) != 1) {
+    std::cerr << "wolfSSL_CTX_use_PrivateKey_file: "
+              << wolfSSL_ERR_error_string(wolfSSL_ERR_get_error(), nullptr) << std::endl;
+    return -1;
+  }
+
+  if (wolfSSL_CTX_use_certificate_chain_file(ssl_ctx_, cert_file) != 1) {
+    std::cerr << "wolfSSL_CTX_use_certificate_chain_file: "
+              << wolfSSL_ERR_error_string(wolfSSL_ERR_get_error(), nullptr) << std::endl;
+    return -1;
+  }
+
+  if (wolfSSL_CTX_check_private_key(ssl_ctx_) != 1) {
+    std::cerr << "wolfSSL_CTX_check_private_key: "
+              << wolfSSL_ERR_error_string(wolfSSL_ERR_get_error(), nullptr) << std::endl;
+    return -1;
+  }
+
+  wolfSSL_CTX_set_session_id_context(ssl_ctx_, sid_ctx, sizeof(sid_ctx) - 1);
+
+  if (config.verify_client) {
+    wolfSSL_CTX_set_verify(ssl_ctx_,
+                           WOLFSSL_VERIFY_PEER | WOLFSSL_VERIFY_CLIENT_ONCE |
+                           WOLFSSL_VERIFY_FAIL_IF_NO_PEER_CERT,
+                           verify_cb);
+  }
+
+  return 0;
+}
+
+extern std::ofstream keylog_file;
+
+namespace {
+void keylog_callback(const SSL *ssl, const char *line) {
+  keylog_file.write(line, strlen(line));
+  keylog_file.put('\n');
+  keylog_file.flush();
+}
+} // namespace
+
+void TLSServerContext::enable_keylog() {
+#ifdef HAVE_SECRET_CALLBACK
```

Comment:
It looks like this macro is not set anywhere.

Reply:
see above.